### ISSUE TITLE
[Bug Report] Open handler gives empty list of open items for async fetchChildren function

### ISSUE THREAD

**Issue by choofy** [vuetifyjs/vuetify#9714]
### Environment
**Vuetify Version:** 2.1.10
**Vue Version:** 2.6.10
**Browsers:** Chrome 77.0.3865.120
**OS:** Windows 10

### Steps to reproduce
See codepen. First click on expand icon next to "A". Then wait until "A1" appears. Then click on expand icon next to "A1". Then, the handleOpen is invoked with no items in the "open" array, which is incorrect: even though A1 is still loading, A is still open.

### Expected Behavior
The array of actual open items is returned.

### Actual Behavior
The array is empty when first event is fired (when loading of children is started), and no event is emitted when the async fetchChildren function actually completes.

### Reproduction Link
<a href="https://codepen.io/choofy01/pen/WNNLdgj" target="_blank">https://codepen.io/choofy01/pen/WNNLdgj</a>

### Other comments
The fix (changes between >> and <<: 

updateOpen(key, isOpen>>, emit = false<<) {
      if (!this.nodes.hasOwnProperty(key)) return;
      const node = this.nodes[key];
      const children = getObjectValueByPath(node.item, this.itemChildren);
      if (children && !children.length && node.vnode && !node.vnode.hasLoaded) {
        node.vnode.checkChildren().then(() => this.updateOpen(key, isOpen, >>true<<));
      } else if (children && children.length) {
        node.isOpen = isOpen;
		console.log('UPDATING OPEN', key, isOpen);
        node.isOpen ? this.openCache.add(key) : this.openCache.delete(key);
        this.updateVnodeState(key);
		>>if (emit) {
			this.emitOpen();
		}<<
      }
    },

**Comment by choofy** [comment]
Note: My proposed fix may not fix everything. I see now that there are 2 issues:

1. When opening A1, an empty "open" array is included in the event, which is incorrect because A is still open. This is not yet covered in my proposed fix.
2. After the children are actually loaded, no open-event is emitted. That should be covered in my proposed fix.

**Comment by choofy** [comment]
One last thing (I hope): I sometimes need the following patch. It may or may not be related to this issue. 

register(node) {
      this.buildTree([node.item]);  <-- this line I had to add.

It has to do with after adding a new child, it is not part of the node hierarchy, which gives an error.

**Comment by KaelWD** [comment]
Treeview items need a unique identifier, if you add `item-key="text"` or an `id` key to each object then it works correctly.

**Comment by KaelWD** [comment]
This could be indicated in the documentation, there isn't really any information other than the examples.

**Comment by choofy** [comment]
I have adjusted the codepen. I made it too simple, apologies for that. The issue occurs when **programmatically** changing the "open" values. In the updated codepen, I do this by responding to the "activate" event, and then adding the active item to the "open" list. In that case, the received "open" array is empty (so, the resulting change of programmatically adding a new item op "open" it not notified back to the application by means of an open-event).
